spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -221,10 +238,36 @@ class CassandraTableScanRDD[R] private[connector](
     if (containsPartitionKey(where)) {
       CassandraPartitionGenerator(connector, tableDef, Some(1), splitSize)
     } else {
-      CassandraPartitionGenerator(connector, tableDef, splitCount, splitSize)
+      CassandraPartitionGenerator(connector, tableDef, numPartitions.orElse(splitCount), splitSize)
     }
   }
 
+  /**
+    * The method does not create CoalesceRDD but reduce number of parittions to read from Cassandra.
+    * It turn off partition size calcutlation and ignore spark.cassandra.input.split.size
+    * The method is useful with where() method call, when actual size of data is smaller then the table size.
+    * It has no effect if a partition key is used in where clause.
+    *
+    * @param numPartitions number of partinions
+    * @param shuffle       weither to call shufle after
+    * @param ord
+    * @return new CassandraTableScanRDD with predefined nube of paritions
+    */
+
+  override def coalesce(numPartitions: Int, shuffle: Boolean = false)(implicit ord: Ordering[R] = null)
+  : RDD[R] = {
+    val rdd = copy(numPartitions = Some(numPartitions))
```

Comment:
I think this should be a `copy(readConf = readConf.copy(splitCount = Some(numPartitions))`